internal/compress_and_encrypt.go

```diff
@@ -17,8 +17,8 @@ type CompressAndEncryptError struct {
 	error
 }
 
-func newCompressingPipeWriterError(reason string) CompressAndEncryptError {
-	return CompressAndEncryptError{errors.New(reason)}
+func newCompressingPipeWriterError(reason string, cause error) CompressAndEncryptError {
+	return CompressAndEncryptError{errors.Wrap(cause, reason)}
```

Comment:
Let's rewrite this to: when `cause` is `nil` - use `errors.New` otherwise `errors.Wrap`... 
It doesn't matter for now, but can lead to cryptic errors in future, when someone pass `nil` to this constrictor - `Wrap` will return nil as well =/
https://github.com/pkg/errors/blob/master/errors.go#L185-L187